tests/test_physics/test_fock/test_fock_repr.py

```diff
@@ -0,0 +1,152 @@
+import numpy as np
+import pytest
+
+import mrmustard.math as math
+from mrmustard.physics.representations import Fock
+from mrmustard.physics.ansatze import ArrayAnsatz
+
+
+class TestFockRepresentation:
+    r"""Tests the init and other algebras related to Fock Representation."""
+
+    def test_init_batched(self):
+        r"""Tests one can initilize a Fock representation with a batched array."""
+        array = math.astensor(np.random.random((1, 5, 7, 8)))
+        fock = Fock(array, batch_flag=True)
+        assert isinstance(fock, Fock)
+        assert np.allclose(fock.array, array)
+
+    def test_init_non_batched(self):
+        r"""Tests one can initilize a Fock representation with an array non batch, and the batch is added after init."""
+        array = math.astensor(np.random.random((5, 7, 8)))
```

Comment:
Why not moving the `math.astensor` inside the init? So users won't need to call it every time